qemu/tests/uefi_check_debugcon.py

```diff
@@ -0,0 +1,114 @@
+    :param env: Dictionary with test environment.
+    """
+
+    def check_trace_process():
+        """
+        check whether trace process is existing
+        """
+        if process.system(
+                params["grep_trace_cmd"], ignore_status=True, shell=True):
+            return False
+        else:
+            return True
+
+    def remove_isa_debugcon(vm):
+        """
+        remove the existing isa-log device
+        """
+        for device in vm.devices:
+            if device.type == "isa-log":
+                vm.devices.remove(device)
+                env.register_vm(vm.name, vm)
+                break
+
+    def trace_kvm_pio():
+        """
+        trace event kvm_pio
+        """
+        process.system(trace_record_cmd)
+
+    error_context.context("Check whether trace-cmd has been "
+                          "installed on host.", logging.info)
+    if process.system("which trace-cmd", ignore_status=True):
+        test.cancel("Please install trace-cmd on host.")
+    if params.get("ovmf_log"):
+        error_context.context("Append debugcon parameter to "
+                              "qemu command lines.", logging.info)
+        ovmf_log = utils_misc.get_path(test.debugdir, params["ovmf_log"])
+        params["extra_params"] %= ovmf_log
+        params["start_vm"] = "yes"
+        env_process.process(test, params, env,
+                            env_process.preprocess_image,
+                            env_process.preprocess_vm)
+    trace_output_file = utils_misc.get_path(test.debugdir,
+                                            params["trace_output"])
+    trace_record_cmd = params["trace_record_cmd"] % trace_output_file
+    check_pio_read = params["check_pio_read"] % trace_output_file
+    check_pio_write = params["check_pio_write"] % trace_output_file
+    stop_trace_record = params["stop_trace_record"]
+    timeout = int(params.get("timeout", 120))
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    error_context.context("Remove the existing isa-log device.", logging.info)
+    remove_isa_debugcon(vm)
+    vm.destroy()
+    error_context.context("Run trace record command on host.",
+                          logging.info)
+    bg = utils_test.BackgroundTest(trace_kvm_pio, ())
+    bg.start()
+    if not utils_misc.wait_for(lambda: bg.is_alive, timeout):
+        test.fail("Failed to start command: '%s'" % trace_record_cmd)
+    try:
+        vm.create()
+        vm.verify_alive()
+        vm.destroy()
+        process.system(stop_trace_record, ignore_status=True, shell=True)
+        if not utils_misc.wait_for(
+                lambda: not check_trace_process(), timeout, 30, 3):
+            test.fail("Failed to stop command: '%s' after %s seconds."
+                      % (stop_trace_record, timeout))
+        pio_read_counts = process.run(
+                check_pio_read, shell=True).stdout.decode().strip()
+        if not eval(pio_read_counts + params["pio_read_counts"]):
```

Comment:
don't do this, please count  pio_read_counts in new line.

Reply:
test.assertLessEqual(pio_read_count, 0)

Reply:
updated, thanks.